src/ui/macstr_ui.py

```diff
@@ -0,0 +1,201 @@
+import sys
+import yaml
+from PyQt5 import QtCore, QtGui, QtWidgets
+
+
+class Ui_MainWindow(object):
+    def setupUi(self, MainWindow):
+
+        # loading in the JSON data
+        with open("../../notebooks/test_image_coords.json", "r") as contents:
+            image_data = yaml.safe_load(contents)
+            print(image_data["cell_image_1.tiff"])
+
+        # creating main GUI window
+        MainWindow.setObjectName("MainWindow")
+        MainWindow.resize(800, 600)
+
+        self.centralwidget = QtWidgets.QWidget(MainWindow)
+        self.centralwidget.setObjectName("centralwidget")
+
+        # ------------------------------
+        # setting up image canvas
+        # ------------------------------
+        self.image1 = QtWidgets.QLabel(self.centralwidget)
+        self.image1.setGeometry(QtCore.QRect(30, 10, 221, 191))
+        self.image1.setText("")
+        self.image1.setPixmap(
+            QtGui.QPixmap("7-stunning-cell-images-from-2017-294838-640x360.jpg")
+        )
+        self.image1.setScaledContents(True)
```

Comment:
How might image data size effect scalable visibility and widget performance? For example, when attempting to scale many large images in this way, would the audience be able to see what they need to see to make decisions? Similarly, would PyQt experience any performance issues when trying to render a large scaled down image in this way?

Reply:
Images being loaded into the GUI will be cropped before hand. Currently, we set the image cropping algorithm to have a size of 256 by 256. I do agree that scaling large images will greatly effect the quality and perfromance of the model. I would think this would be up to the user judgement indicating how their images should be cropped before loading into the application.

Reply:
Thanks, understood, and sounds good.